scripts/sanity_check_test.py

```diff
@@ -0,0 +1,139 @@
+import unittest
+import sys
+
+import sanity_check
+import git
+
+class ValidateTests(unittest.TestCase):
+    """ Tests for commit validation """
+
+    def assertFailure(self, commits, expected):
+        from StringIO import StringIO
+
+        real_commits = convert_commits(commits)
+        out = StringIO()
+
+        self.assertRaises(SystemExit, sanity_check.validate_commits, real_commits, out)
+        assert out.getvalue().strip() == expected
+
+    def assertSuccess(self, commits):
+        real_commits = convert_commits(commits)
+
+        try:
+            sanity_check.validate_commits(real_commits)
+        except:
+            print "Unexpected error:", sys.exc_info()[0]
+            raise
+
+
+    # Test cases that excercise invalid scenarios.
+
+    def test_squash_not_matching_carry(self):
+        """
+        [SQUASH] commit has a type that is not matching any [CARRY].
+        """
+
+        commits = [
+        "pick c21ea24 [CARRY][BRANDING] OAuth server branding",
+        "pick c21ea21 [SQUASH][BRANDING] OAuth server branding",
+        "pick c21ea22 [SQUASH][DIFFERENT] Different kind of stuff",
+        "pick 975a829 [CARRY][BUILD_GEN] Specfile updates",
+        "pick ba426b9 Automatic commit of package [atomic-openshift] release [3.5.0.50-1].",
+        ]
+        expected = "Not all SQUASH commits correspond to a CARRY commit!"
+        self.assertFailure(commits, expected) 
+
+    def test_carry_without_type(self):
+        """
+        [CARRY] commit does not have a type.
+        """
+
+        commits = [
+        "pick c21ea24 [CARRY][BRANDING] OAuth server branding",
+        "pick c21ea21 [SQUASH][BRANDING] OAuth server branding",
+        "pick 975a829 [CARRY][BUILD_GEN] Specfile updates",
+        "pick ba426b9 Automatic commit of package [atomic-openshift] release [3.5.0.50-1].",
+        "pick d5f4ssw [DROP] because it is temporary",
+        "pick 975a829 [CARRY] without a type because I am lazy",
+        ]
+        expected = "Found a [CARRY] commit without a type: `[CARRY] without a type because I am lazy` (975a829)"
+        self.assertFailure(commits, expected)
+
+    def test_squash_without_type(self):
+        """
+        [SQUASH] commit does not have a type.
+        """
+
+        commits = [
+        "pick c21ea24 [CARRY][BRANDING] OAuth server branding",
+        "pick c21ea21 [SQUASH][BRANDING] OAuth server branding",
+        "pick c21ea22 [SQUASH][DIFFERENT] Different kind of stuff",
+        "pick 975a829 [CARRY][BUILD_GEN] Specfile updates",
+        "pick ba426b9 Automatic commit of package [atomic-openshift] release [3.5.0.50-1].",
+        "pick 975a829 [SQUASH] without a type because I am lazy",
+        ]
+        expected = "Found a [SQUASH] commit without a type: `[SQUASH] without a type because I am lazy` (975a829)"
+        self.assertFailure(commits, expected)
+
+    def test_missing_tito_tag(self):
```

Comment:
All the logic is found in validate_commits. Are you suggesting spliting commit message validation out of cross-commit validation?